api/proto/teleport/devicetrust/v1/device.proto

```diff
@@ -93,7 +93,32 @@ message DeviceCredential {
   // Unique identifier of the credential, defined client-side.
   string id = 1;
   // Device public key marshaled as a PKIX, ASN.1 DER.
+  // For TPMs, this is the public application key.
   bytes public_key_der = 2;
+  // The degree to which the device credential is attested.
+  AttestationType attestation_type = 3;
+  // If the device is a TPM device, stores the serial number from the TPM
+  // endorsement certificate.
+  // The certificate is not stored for security reasons, but the serial can be
+  // used to retrieve information about a specific unit from the manufacturer
+  // at a later date.
+  string tpm_serial = 4;
+  // If the device is a TPM device, stores the public AK in PKIX, ASN.1 DER
+  // form.
+  bytes tpm_attestation_key_der = 5;
+}
+
+// AttestationType indicates the degree to which the device credential has
+// been attested.
+enum AttestationType {
+  // Bare public key which has only verified with proof of ownership.
+  // Used on macOS.
+  ATTESTATION_TYPE_UNSPECIFIED = 0;
+  // Credential was verified through an EK->AK->App key chain on enrollment.
+  ATTESTATION_TYPE_HW_KEY = 1;
+  // Credential was verified through an EK->AK->App key chain on enrollment and
+  // an EKCert was present and signed by a configured approved TPM CA.
+  ATTESTATION_TYPE_HW_CERT = 2;
```

Comment:
Maybe spell it out?

```suggestion
  ATTESTATION_TYPE_HARDWARE_KEY = 1;
  // Credential was verified through an EK->AK->App key chain on enrollment and
  // an EKCert was present and signed by a configured approved TPM CA.
  ATTESTATION_TYPE_HARDWARE_CERT = 2;
```

It could be worth saying "TPM" explicitly too.

Reply:
I actually changed this to include TPM but then undid it because I didn't want to deviate from the spec - so actually very happy to include TPM and spell them out fully - I'm not a fan of shortening words :)

Reply:
Deviate as much as you like, this here is the real deal. We can update the RFD later, although I wouldn't worry too much about a few renames, it's bound to happen.

Reply:
DeviceAttestationType